app/src/main/java/com/example/travellens/fragments/DetailFragment.java

```diff
@@ -191,38 +164,106 @@ public void onClick(View v) {
         });
     }
 
-    private void likeOrUnlike() {
-        // checks whether user has previously liked this post
-        // we want to check on every onClick
-        List<String> list = ParseUser.getCurrentUser().getList("savedPosts");
-        ArrayList<String> list1 = new ArrayList<String>();
-        if (list != null)
-            list1 = new ArrayList<String>(list);
-        boolean alreadyLiked = false;
-        if (list1.contains(thePost.getObjectId())) {
-            alreadyLiked = true;
-        }
-        if (alreadyLiked) {
-            ivLikes.setImageResource(R.drawable.ufi_heart);
-            list1.remove(thePost.getObjectId());
-            ParseUser.getCurrentUser().put("savedPosts", list1);
-            thePost.put("likes", thePost.getNumber("likes").intValue() - 1);
-        } else {
-            ivLikes.setImageResource(R.drawable.img_2);
-            list1.add(thePost.getObjectId());
-            ParseUser.getCurrentUser().put("savedPosts", list1);
-            thePost.put("likes", thePost.getNumber("likes").intValue() + 1);
-        }
+    private void goToProfile() {
+        ProfileFragment profileFragment = new ProfileFragment(thePost.getUser());
+        AppCompatActivity activity = (AppCompatActivity)getActivity();
+        activity.getSupportFragmentManager().beginTransaction().replace(R.id.flContainer, profileFragment).addToBackStack(null).commit();
+    }
 
-        thePost.saveInBackground(new SaveCallback() {
+    private void likeOrUnlike() {
+        ParseQuery<Likes> query = ParseQuery.getQuery(Likes.class);
+        query.whereEqualTo(Likes.KEY_POST, thePost);
+        query.whereEqualTo(Likes.KEY_USER, ParseUser.getCurrentUser());
+        query.setLimit(1);
+        query.findInBackground(new FindCallback<Likes>() {
             @Override
-            public void done(ParseException e) {}});
-        ParseUser.getCurrentUser().saveInBackground(new SaveCallback() {
+            public void done(List<Likes> results, ParseException e) {
+                if (e == null && !results.isEmpty()) {
```

Comment:
This doesn't handle if there is an actual exception. The else essentially ignores the exception.